core/src/main/java/kafka/server/ReplicaFetcherTierStateMachine.java

```diff
@@ -229,6 +229,9 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
                 Partition partition = replicaMgr.getPartitionOrException(topicPartition);
                 partition.truncateFullyAndStartAt(nextOffset, false, Option.apply(leaderLogStartOffset));
 
+                replicaMgr.brokerTopicStats().topicStats(topicPartition.topic()).buildRemoteLogAuxStateRequestRate().mark();
+                replicaMgr.brokerTopicStats().allTopicsStats().buildRemoteLogAuxStateRequestRate().mark();
+
```

Comment:
Question: Why do we update request rate metric here? 
I'm thinking we could mark the request rate and error rate at the caller of `buildRemoteLogAuxState` method. That is, in `ReplicaFetcherTierStateMachine#start`

```
long offsetToFetch = 0;
// mark request rate here
try {
  offsetToFetch = buildRemoteLogAuxState(topicPartition, currentFetchState.currentLeaderEpoch(), leaderLocalStartOffset, epoch, fetchPartitionData.logStartOffset());
} catch (RemoteStorageException e) {
  // mark error rate here
  throw e;
}
```

This way, if there are exception thrown from other places, like in L219 `fetchRemoteLogSegmentMetadata` method, we can still catch that and update the metrics. WDYT?

Reply:
The original reason was not to emit the error metrics, when buildRemoteLogAuxState fails because of the remote log storage not being enabled. However, thinking about this again, this makes more sense. We will always emit request metrics when we attempt to build log aux, and emit error metrics when there is an exception. 

I made the suggested change.